pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/SegmentMetadataImpl.java

```diff
@@ -94,18 +96,46 @@
   private long _segmentStartTime;
   private long _segmentEndTime;
 
-  public SegmentMetadataImpl(File indexDir)
+  public SegmentMetadataImpl(@Nonnull File indexDir, @Nonnull SegmentVersion segmentVersion)
       throws ConfigurationException, IOException {
-    LOGGER.debug("SegmentMetadata location: {}", indexDir);
-    File tmpMetadataFile = SegmentDirectoryPaths.findMetadataFile(indexDir);
-    if (tmpMetadataFile == null) {
-      LOGGER.warn("Metadata file: {} does not exist in directory: {}", tmpMetadataFile, indexDir);
-      tmpMetadataFile = new File(indexDir, MetadataKeys.METADATA_FILE_NAME);
+    _metadataFile = SegmentDirectoryPaths.findMetadataFile(indexDir, segmentVersion);
+    Preconditions.checkNotNull(_metadataFile, "Cannot find segment metadata file under directory: " + indexDir);
```

Comment:
Older code allowed non-existing (empty?) segment metadata file. Not sure if that will be regression - check SegmentFetcherAndLoader and segment generator code for any dependency.